app/controllers/auth/actions/VatUserAction.scala

```diff
@@ -14,21 +14,19 @@
  * limitations under the License.
  */
 
-package controllers
+package controllers.auth.actions
 
-import javax.inject.{Inject, Singleton}
-
-import config.AppConfig
-import play.api.i18n.{I18nSupport, MessagesApi}
+import controllers.auth.{AuthPredicates, AuthorisedActions}
 import play.api.mvc.{Action, AnyContent}
 import uk.gov.hmrc.play.frontend.controller.FrontendController
 
+import scala.concurrent.Future
 
-@Singleton
-class UnauthorisedController @Inject()(val messagesApi: MessagesApi,
-                                       implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
+trait VatUserAction extends AuthorisedActions {
+  self: FrontendController =>
 
-  val show: Action[AnyContent] = Action { implicit request =>
-    Ok(views.html.unauthorised())
+  object VatUserAction {
+    def apply(action: ActionBody): Action[AnyContent] = async(action andThen (_ andThen Future.successful))
```

Comment:
I don't think this is needed if you just call `VatUserAction.async` whenever you need it